components/badge/index.jsx

```diff
@@ -0,0 +1,145 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+// # Badges Component
+
+// Implements the [Badges design pattern](https://www.lightningdesignsystem.com/components/badges/) in React.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import requiredIf from 'react-required-if';
+import classNames from 'classnames';
+
+import BadgeIcon from '../icon/badge-icon';
+
+import { BADGE } from '../../utilities/constants';
+
+const propTypes = {
+	/**
+	 * Change badge color and that of text accordingly
+	 */
+	color: PropTypes.oneOf(['inverse', 'lightest']),
+	/**
+	 * CSS classes that are applied to the span.
+	 */
+	className: PropTypes.oneOfType([
+		PropTypes.array,
+		PropTypes.object,
+		PropTypes.string,
+	]),
+	/**
+	 * Name of the icon category. Visit <a href="http://www.lightningdesignsystem.com/resources/icons">Lightning Design System Icons</a> to reference icon categories.
+	 */
+	iconCategory: requiredIf(
```

Comment:
Many of the older components do use `iconCategory` and `iconName` but the best pattern is to pass in a whole `<Icon>` so that the `Icon` props do not have to be duplicated within badge also https://github.com/salesforce/design-system-react/blob/master/docs/codebase-overview.md#reuse-existing-component-props-by-using-component-no-button-iconclassname-

It looks like there are special classnames for badge that are on placed on the icon, so you may want to duplicate the icon passed into a prop and the classNames that way before render.

`iconPosition: left|right` is how `Button` currently works. 

Please update.